## Task 15 — Customer Count by Country

### Business Question

Which countries have the largest customer base?

### Analysis

Group customers by country, calculate the number of customers for each country, and sort the results in descending order by customer count.

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
customer_count_by_country = starlink.groupby("country").agg(
    users_amount = ("unique_id","count"),
).sort_values(
    "users_amount",
    ascending = False
)
print(customer_count_by_country)

                users_amount
country                     
Ukraine                 1032
Australia               1030
Germany                 1027
Spain                   1019
Canada                   997
Brazil                   996
France                   996
United Kingdom           977
Japan                    973
United States            954


### Result

Ukraine has the largest customer base with 1,032 customers, followed by Australia with 1,030 and Germany with 1,027 customers. The United States has the smallest customer base with 954 customers.

### Business Insight

Customer distribution is relatively balanced across the analyzed countries, with the difference between the largest and smallest customer bases being relatively small. Ukraine has the largest customer base, while the United States has the smallest.